### PR TITLE
[multi asic] Change the fib test to support multi asic platform

### PR DESCRIPTION
**Pull request by arlakshm**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ x] Test case(new/improvement)

#### What is the motivation for this PR?
The PR has changes to run the  tests in `test_fib.py` on multi asic platforms

#### How did you do it?
The PR has following changes.
 - Update the fixtures in the `test_fib.py` for multi asic platforms
 - Add a new options `ignore_ttl` to the ptf fib tests.
 
#### How did you verify/test it?
Verify on multi asic

```
arlakshm@b65425d3c0c1:/data/repos/public/sonic-mgmt/tests$ ./run_tests.sh -d multi_asic_1 -n vms13-t1-lag-2 -c fib/test_fib.py -i /data/repos/files/veos_vtb -f /data/repos/files/vtestbed.csv -l WARNING -k debug -u -m individual -t t1,any -e '--deep_clean --disable_loganalyzer --skip_sanity '
=== Running tests individually ===
/usr/local/lib/python2.7/dist-packages/ansible/parsing/vault/__init__.py:44: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in a future release.
  from cryptography.exceptions import InvalidSignature
================================================================================= test session starts ==================================================================================
platform linux2 -- Python 2.7.17, pytest-4.6.5, py-1.9.0, pluggy-0.13.1
ansible: 2.8.12
rootdir: /data/repos/public/sonic-mgmt/tests, inifile: pytest.ini
plugins: forked-1.3.0, xdist-1.28.0, html-1.22.1, metadata-1.10.0, repeat-0.9.1, ansible-2.2.2
collected 3 items

fib/test_fib.py::test_basic_fib[True-True-1514] PASSED                                                                                                                           [ 33%]
fib/test_fib.py::test_hash[ipv4] PASSED                                                                                                                                          [ 66%]
fib/test_fib.py::test_hash[ipv6] PASSED                                                                                                                                          [100%]

---------------------------------------------------- generated xml file: /data/repos/public/sonic-mgmt/tests/logs/fib/test_fib.xml -----------------------------------------------------
============================================================================= 3 passed in 1818.33 seconds ==============================================================================
+ exit 0
arlakshm@b65425d3c0c1:/data/repos/public/sonic-mgmt/tests$

```

### CODE REVIEW

File: tests/fib/test_fib.py [hunk 1]
```diff
@@ -67,41 +76,55 @@ def get_fib_info(duthost, cfg_facts, mg_facts):
...
-                if po.has_key(ifname):
-                    oports.append([str(mg_facts['minigraph_ptf_indices'][x]) for x in po[ifname]['members']])
-                else:
-                    if ports.has_key(ifname):
-                        oports.append([str(mg_facts['minigraph_ptf_indices'][ifname])])
+    for asic_index, asic_cfg_facts in  enumerate(cfg_facts):
+
+        asic = duthost.asic_instance(asic_index)
+
+        asic.shell("{} redis-dump -d 0 -k 'ROUTE*' -y > /tmp/fib.{}.txt".format(asic._ns_arg, timestamp))
```
Comment (smaheshm) [inline tests/fib/test_fib.py:84]: why not use "asic.command()". No need to use "asic._ns_arg" which looks like a private attribute as per convention.

File: tests/common/devices/sonic_asic.py [hunk 2]
```diff
@@ -115,6 +115,7 @@ def config_facts(self, *module_args, **complex_args):
             complex_args['namespace'] = self.namespace
         return self.sonichost.config_facts(*module_args, **complex_args)
 
+
```
Comment (smaheshm) [inline tests/common/devices/sonic_asic.py:118]: nit: remote extra space

File: tests/fib/test_fib.py [hunk 3]
```diff
@@ -67,41 +76,55 @@ def get_fib_info(duthost, cfg_facts, mg_facts):
...
+                ifnames = v['value']['ifname'].split(',')
+                nh = v['value']['nexthop']
+
+                oports = []
+                for ifname in ifnames:
+                    if po.has_key(ifname):
+                        # ignore the prefix, if the prefix nexthop is not a frontend port
+                        if 'members' in po[ifname]:
+                            if 'role' in ports[po[ifname]['members'][0]] and ports[po[ifname]['members'][0]]['role'] == 'Int':
+                                skip = True
```
Comment (smaheshm) [inline tests/fib/test_fib.py:105]: Are we checking if prefixes from T0 ASICs are learned on T1 ASICS, and vice versa? We are removing all the prefixes that have next hop as the backend.

Comment (arlakshm) [reply tests/fib/test_fib.py:105]: @smaheshm for this test purpose there is no difference if the route is learnt from T0 or T2.  We need to build the fib table with all the external nexthops for the ptf tests to validate the packet forwarding

File: tests/fib/test_fib.py (hunk 1 above)
Comment (arlakshm) [reply tests/fib/test_fib.py]: @smaheshm  the ansible command module doesnt work with `>` 
https://github.com/ansible/ansible/blob/f9f839fa08eee46ad7a86d6cbc7519541a50c7ef/lib/ansible/modules/command.py#L22. `shell` module is recommended for such cases

Change the `ns_arg` to public variable

File: tests/common/devices/sonic_asic.py (hunk 2 above)
Comment (arlakshm) [reply tests/common/devices/sonic_asic.py]: fixed

File: tests/fib/test_fib.py (hunk 3 above)
Comment (arlakshm) [reply tests/fib/test_fib.py:105]: We use -64-lag topology for multi asic platform testing. In this topology some ASICs T2 neighbors and some have T0 neighbors.  This covers the case of more than one ASIC having the same prefix.

**Comment by arlakshm** [comment]
@smaheshm , @wangxin  can you review this PR and approve if not more comments

**Review by smaheshm — APPROVED** [review]
:shipit:

**Review by wangxin — APPROVED** [review]
LGTM, @arlakshm, can you address the merge conflicts?

**Pull request merged**